### PR TITLE
Disqualify `auto trait` built-in impl in new solver if explicit `impl` exists

### PR DESCRIPTION
**Pull request by compiler-errors**

### CODE REVIEW

File: compiler/rustc_trait_selection/src/solve/trait_goals.rs [hunk 1]
```diff
@@ -89,6 +89,15 @@ impl<'tcx> assembly::GoalKind<'tcx> for TraitPredicate<'tcx> {
         ecx: &mut EvalCtxt<'_, 'tcx>,
         goal: Goal<'tcx, Self>,
     ) -> QueryResult<'tcx> {
+        if let Some(def_id) = ecx.tcx().find_map_relevant_impl(
```
Comment (lcnr) [inline compiler/rustc_trait_selection/src/solve/trait_goals.rs:92]:
```suggestion
        // This differs from the current stable behavior and
        // fixes #84857. Due to breakage found via crater, we
        // currently instead lint patterns which can be used to
        // exploit this unsoundness on stable, see #93367 for
        // more details.
        if let Some(def_id) = ecx.tcx().find_map_relevant_impl(
```
context is good i think

**Review by lcnr — COMMENTED** [review]
r=me after added comment

File: compiler/rustc_trait_selection/src/solve/trait_goals.rs (hunk 1 above)
Comment (compiler-errors) [reply compiler/rustc_trait_selection/src/solve/trait_goals.rs:97]: Whoops, yeah, sorry.

**Pull request merged**